packages/cactus-test-tooling/src/main/typescript/quorum/quorum-mp-test-ledger.ts

```diff
@@ -0,0 +1,255 @@
+import { EventEmitter } from "events";
+import Docker, { Container, ContainerCreateOptions } from "dockerode";
+import {
+  Logger,
+  LoggerProvider,
+  LogLevelDesc,
+} from "@hyperledger/cactus-common";
+import { ITestLedger } from "../i-test-ledger";
+import { Containers } from "../common/containers";
+
+export interface IQuorumMultiPartyTestLedgerOptions {
+  readonly containerImageName?: string;
+  readonly containerImageVersion?: string;
+  readonly logLevel?: LogLevelDesc;
+  readonly emitContainerLogs?: boolean;
+  readonly envVars?: string[];
+  // For test development, attach to ledger that is already running, don't spin up new one
+  readonly useRunningLedger?: boolean;
+}
+
+export class QuorumMultiPartyTestLedger implements ITestLedger {
+  public readonly containerImageName: string;
+  public readonly containerImageVersion: string;
+  private readonly logLevel: LogLevelDesc;
+  private readonly emitContainerLogs: boolean;
+  private readonly useRunningLedger: boolean;
+  private readonly envVars: string[];
+
+  private readonly log: Logger;
+  public container: Container | undefined;
+  public containerId: string | undefined;
+
+  constructor(public readonly options?: IQuorumMultiPartyTestLedgerOptions) {
+    // @todo Replace with hyperledger ghcr link when available
+    this.containerImageName =
+      options?.containerImageName ||
+      "ghcr.io/outsh/cactus-quorum-multi-party-all-in-one";
+    this.containerImageVersion =
+      options?.containerImageVersion || "2022-03-14-65afd20";
+    this.logLevel = options?.logLevel || "info";
+
+    if (options?.emitContainerLogs != undefined) {
+      this.emitContainerLogs = options.emitContainerLogs;
+    } else {
+      this.emitContainerLogs = true;
+    }
+
+    if (options?.useRunningLedger != undefined) {
+      this.useRunningLedger = options.useRunningLedger;
+    } else {
+      this.useRunningLedger = false;
+    }
+
+    this.envVars = options?.envVars || [];
+
+    this.log = LoggerProvider.getOrCreate({
+      level: this.logLevel,
+      label: "quorum-multi-party-test-ledger",
+    });
+  }
+
+  public get fullContainerImageName(): string {
+    return [this.containerImageName, this.containerImageVersion].join(":");
+  }
+
+  public async start(omitPull = false): Promise<Container> {
+    if (this.useRunningLedger) {
+      this.log.info(
+        "Search for already running Quorum Test Ledger because 'useRunningLedger' flag is enabled.",
+      );
+      const containerInfo = await Containers.getByPredicate(
+        (ci) =>
+          ci.Image === this.fullContainerImageName && ci.State === "running",
+      );
+      const docker = new Docker();
+      this.containerId = containerInfo.Id;
+      this.container = docker.getContainer(this.containerId);
+      return this.container;
+    }
+
+    this.log.info(
```

Comment:
@outSH Continuing on my comment above, this message is good in that it prints the image name, but it only reaches this point in the execution on the happy path not when people would need to figure out what's going wrong.

Reply:
@petermetz Done.